c0isc_demo.c

```diff
@@ -245,63 +232,174 @@ static int ping_input_prepare(struct m0_buf *buf, struct m0_fid *comp_fid,
 }
 
 static int input_prepare(struct m0_buf *buf, struct m0_fid *comp_fid,
-			 uint32_t *reply_len, enum isc_comp_type type,
-			 void *ip_args)
+			 struct m0_layout_io_plop *iop, uint32_t *reply_len,
+			 enum isc_comp_type type)
 {
 	switch (type) {
 	case ICT_PING:
 		return ping_input_prepare(buf, comp_fid, reply_len, type);
 	case ICT_MIN:
 	case ICT_MAX:
-		return minmax_input_prepare(buf, comp_fid, reply_len, type,
-					    ip_args);
+		return minmax_input_prepare(buf, comp_fid, iop,
+					    reply_len, type);
 	}
 	return -EINVAL;
 }
 
-struct mm_result *op_result(struct mm_result *x, struct mm_result *y,
-			    enum isc_comp_type op_type)
+static struct mm_result *
+op_result(struct mm_result *x, struct mm_result *y, enum isc_comp_type op_type)
 {
+	int               rc;
+	int               len;
+	struct mm_result *res;
+	char             *buf;
+	double            x_rval;
+	double            y_lval;
+
+	len = x->mr_rbuf.i_len + y->mr_lbuf.i_len;
+	buf = malloc(x->mr_rbuf.i_len + y->mr_lbuf.i_len + 1);
+	if (buf == NULL) {
+		fprintf(stderr, "failed to allocate %d of memory for result\n",
+			                            len);
+		return NULL;
+	}
+
+	memcpy(buf, x->mr_rbuf.i_buf, x->mr_rbuf.i_len);
+	memcpy(buf + x->mr_rbuf.i_len, y->mr_lbuf.i_buf, y->mr_lbuf.i_len);
+	buf[x->mr_rbuf.i_len + y->mr_lbuf.i_len] = '\0';
+
+	rc = sscanf(buf, "%lf%n", &x_rval, &len);
+	if (rc < 1) {
+		fprintf(stderr, "failed to read the resulting xr-value\n");
+		m0_free(buf);
+		return NULL;
+	}
+
+	y->mr_idx     += x->mr_idx_max;
+	y->mr_idx_max += x->mr_idx_max;
+
+	//printf("buf=%s x_rval=%lf\n", buf, x_rval);
```

Comment:
C, please. The same below.

Reply:
It's convenient to comment out / uncomment quickly with such style of comments. I know it's non-standard one, probably a gcc extension. But aren't we using a lot of gcc extensions anyway? So why not to use such style of comments?